commit 4b00bfdd73a2cd56a4d9e8de0b249eed2a1b982f
Author: petarj <petarj@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Fri Feb 3 00:34:52 2017 +0000

    add suppression for helgrind/tests/tc22_exit_w_lock
    
    Function pthread_create indirectly calls function memcpy. Helgrind
    considers that memcpy is not thread safe function. For error reported
    from pthread_create there is the suppression helgrind---_dl_allocate_tls
    in the file glibc-2.34567-NPTL-helgrind.supp.
    Since glibc version 2.23, memcpy is implemented by __mempcpy_inline.
    This causes that call to memcpy from pthread_create is no longer
    recognized by the suppression.
    In test helgrind/tests/tc22_exit_w_lock, pthread_create is called twice,
    and second call reports error, which causes failing of the test.
    This patch adds suppression for glibc 2.23 and greater.
    
    Patch by Tamara Vlahovic.
    
    Related issue #375806.
    
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@16219 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/glibc-2.34567-NPTL-helgrind.supp b/glibc-2.34567-NPTL-helgrind.supp
index ed105b8..7ebd2c4 100644
--- a/glibc-2.34567-NPTL-helgrind.supp
+++ b/glibc-2.34567-NPTL-helgrind.supp
@@ -267,6 +267,18 @@
    fun:pthread_create@*
 }
 
+{
+   helgrind---_dl_allocate_tls2
+   Helgrind:Race
+   fun:memcpy
+   fun:__mempcpy_inline
+   fun:_dl_allocate_tls_init
+   ...
+   fun:pthread_create@@GLIBC_2.2*
+   fun:pthread_create_WRK
+   fun:pthread_create@*
+}
+
 ####################################################
 # To do with GNU libgomp
 #
